### DISCUSSION TITLE
Here’s how I broke Coolify – while updating through the web interface.

### DISCUSSION THREAD

**Discussion by ooomytech** [coollabsio/coolify#3687] [Improvement Requests]
Hey everyone,
 I just wanted to warn you about the possibility of breaking your Coolify server during an update.
I was trying to update my server and clicked the update button several times because after the first click, the button doesn’t get disabled. It seems like the update script starts running multiple times when you click it again. 

![2024-10-03_13-07-54](https://github.com/user-attachments/assets/1da17d35-f887-498b-a97b-3b3346030ee1)

As a result, my backup .env file wasn’t saved. Something went wrong—the new file wasn’t created, and the old backup file didn’t save. It looks like the repeated clicking caused the update to run multiple times, and some files overwrote others.

In the end, I lost my env settings, and most importantly, I lost the encryption key for my Laravel app. To get it running again, I had to create a new key, but now all the old projects won’t open. I keep getting the error: “Error: The MAC is invalid” for almost all the old applications.

I’m hesitant to try and reproduce this issue again, just wanted to warn you all not to click the update button multiple times. If a developer is reading this, it might be a good idea to disable the button and show that the process has started—right now, that only happens after 3-5 seconds.

-----------------
I also want to tell the developer that the application itself is just amazing. I’m sure you’ve probably heard this many times, but I want to say it again. If I could give 100 stars instead of one, I would! Thank you so much for your work.

**Comment by bartvdbraak** [comment]
I think I just had the same issue..

**Reply by ali-sultani-dev** [reply]
Same here, I just broke mine and am searching for solutions.

**Reply by ignasandr** [reply]
Same. Huge relief that the services are all still up.

**Reply by ignasandr** [reply]
If anyone still has the problem, running /data/coolify/source/upgrade.sh fixes the issue
https://github.com/coollabsio/coolify/issues/4218

**Reply by pencilcheck** [reply]
> If anyone still has the problem, running /data/coolify/source/upgrade.sh fixes the issue #4218

I wish coolify can create a dedicated command to fix similar to homebrew has brew doctor or something

**Reply by Martoxdlol** [reply]
> If anyone still has the problem, running /data/coolify/source/upgrade.sh fixes the issue #4218

You are my hero!!!

**Reply by mhsazol111** [reply]
You are the best!

**Comment by peaklabs-dev** [comment]
This is a know issues and we should fix this asap, there is also an issue about this: https://github.com/coollabsio/coolify/issues/3407

**Reply by Code-DJ** [reply]
> This is a know issues and we should fix this asap, there is also an issue about this: #3407

Does it resolve this? https://github.com/coollabsio/coolify/issues/3890 Should I close that issue?

**Comment by andrasbacsai** [comment]
This will be fixed in the next version.

**Reply by ali-sultani-dev** [reply]
Thanks Andras, I’ve been using Coolify, and I absolutely love it.

**Comment by woutervanwijk** [comment]
Same here. I don't remember pressing the button multiple times. These are the logs:

upgrade-2024-11-13-01-00-05.log:
```
 coolify Pulling
 coolify Error toomanyrequests: retry-after: 394.314µs, allowed: 44000/minute
Error response from daemon: toomanyrequests: retry-after: 394.314µs, allowed: 44000/minute
```

upgrade-2024-11-13-15-40-06.log
```
 Container coolify-db  Recreate
 Container coolify-realtime  Recreate
 Container coolify-redis  Recreate
 Container coolify-redis  Recreated
 Container coolify-db  Recreated
 Container coolify-realtime  Recreated
 Container coolify  Recreate
Error response from daemon: No such container: 532df57a9abf60721c90b1b97dcd0a35d04ee4565f0da32f643b24910065cf88
```

upgrade-2024-11-13-15-40-08.log

```
 Container coolify-db  Recreate
 Container coolify-realtime  Recreate
 Container coolify-redis  Recreate
 Container coolify-db  Recreated
 Container coolify-realtime  Recreated
 Container coolify-redis  Recreated
 Container ef34891eeeba_coolify  Stopping
 Container coolify  Recreate
 Container ef34891eeeba_coolify  Stopped
 Container ef34891eeeba_coolify  Removing
 Container ef34891eeeba_coolify  Removed
Error response from daemon: Conflict. The container name "/ef34891eeeba_coolify" is already in use by container "532df57a9ab
f60721c90b1b97dcd0a35d04ee4565f0da32f643b24910065cf88". You have to remove (or rename) that container to be able to reuse that name.

```

I'm glad to report that running  /data/coolify/source/upgrade.sh indeed fixes the issue, thanks @ignasandr !


Thanks for Coolify, it's 😎

**Comment by sovetski** [comment]
@ooomytech as @ignasandr suggested here:

> If anyone still has the problem, running /data/coolify/source/upgrade.sh fixes the issue #4218

It looks like the problem is solved and this information can be useful for everyone, can you make this discussion as "solved" ? That will propbably help other people in the future and keep discussion history clean to filter all old/solved ones with "Answered" label
